main.tf

```diff
@@ -1,7 +1,240 @@
+  task_role_arn            = aws_iam_role.task_role.arn
+
+  tags = var.tags
+}
+
+# ==================== Fargate ====================
+resource "aws_ecs_cluster" "cluster" {
+  count = var.ecs_cluster_arn == null ? 1 : 0
+  name  = local.resource_name
+  tags  = var.tags
+}
+resource "aws_security_group" "fargate_service_sg" {
+  name        = "${local.resource_name}-fargate-sg"
+  description = "Controls access to the Fargate Service"
+  vpc_id      = var.vpc_id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags = var.tags
+}
+
+# ==================== Cloudwatch Event ====================
+# --- CloudWatch Event IAM Role --- TODO see if we can use this role instead of PowerBuilder
+//data "aws_iam_policy_document" "cloudwatch-event-assume-role-policy" {
+//  version = "2012-10-17"
+//  statement {
+//    effect  = "Allow"
+//    actions = ["sts:AssumeRole"]
+//    principals {
+//      identifiers = ["events.amazonaws.com"]
+//      type        = "Service"
+//    }
+//  }
+//}
+//resource "aws_iam_role" "scheduled-task" {
+//  name                 = "${local.resource_name}-scheduled-task-role"
+//  assume_role_policy   = data.aws_iam_policy_document.cloudwatch-event-assume-role-policy.json
+//  permissions_boundary = var.role_permissions_boundary_arn
+//  tags                 = var.tags
+//}
+//data "aws_iam_policy_document" "event-policy" {
+//  version = "2012-10-17"
+//  statement {
+//    effect    = "Allow"
+//    actions   = ["ecs:runTask"]
+//    resources = ["*"]
+//  }
+//  statement {
+//    effect    = "Allow"
+//    actions   = ["iam:PassRole"]
+//    resources = [aws_iam_role.task_execution_role.arn, aws_iam_role.task_role.arn]
+//  }
+//}
+//resource "aws_iam_policy" "cloudwatch-policy" {
+//  name   = "${local.resource_name}-event-policy"
+//  policy = data.aws_iam_policy_document.event-policy.json
+//}
+//resource "aws_iam_role_policy_attachment" "cloudwatch-event-role-policy-attach" {
+//  policy_arn = aws_iam_policy.cloudwatch-policy.arn
+//  role       = aws_iam_role.scheduled-task.name
+//}
+# --- CloudWatch Event Rule ---
+resource "aws_cloudwatch_event_rule" "scheduled_task" {
+  name                = "${local.resource_name}-scheduled-task"
+  description         = "Run ${local.resource_name} task at a scheduled time (${var.schedule_expression})"
+  schedule_expression = var.schedule_expression
+}
+resource "aws_cloudwatch_event_target" "scheduled_task" {
+  target_id = "${local.resource_name}-scheduled-task-target"
+  rule      = aws_cloudwatch_event_rule.scheduled_task.name
+  arn       = var.ecs_cluster_arn == null ? aws_ecs_cluster.cluster[0].arn : var.ecs_cluster_arn
+  //  role_arn  = aws_iam_role.scheduled-task-cloudwatch.arn
+  role_arn = var.event_role_arn
+
+  ecs_target {
+    task_count          = 1
+    task_definition_arn = aws_ecs_task_definition.task_def.arn
+    launch_type         = "FARGATE"
+    platform_version    = "LATEST"
```

Comment:
Upgrade to 1.4.0 (I believe that is the latest)

Reply:
Done